### DISCUSSION TITLE
Opt in to exhaustive enum handling (UnhandledMatchCondition) without match statements?

### DISCUSSION THREAD

**Discussion by AlbinoDrought** [vimeo/psalm#10059] [Q&A] [answered]
Hello, if we have the following simplified code: https://psalm.dev/r/7f1e92b68e

and in the future we add another value to the enum: https://psalm.dev/r/203dacfe6a

`UnhandledMatchCondition` is emitted for the `exhaustiveMatch` function only. Is it possible to assert that all enum values should be handled in the `exhaustiveSwitch` and `exhaustiveIf` functions as well, so errors are also emitted there?

I'm imagining something like this: https://psalm.dev/r/030f9f52e5

Cheers

**Comment by weirdan** [comment, marked as answer]
I think you can do that already with `@psalm-check-type`: https://psalm.dev/r/5ca4f398f1
